packages/langchain-graph-retriever/src/langchain_graph_retriever/adapters/astra.py

```diff
@@ -37,61 +37,62 @@
 _MAX_RETRIES = 3
 
 
-class _QueryHelper:
-    def __init__(
-        self,
-        codec: _AstraDBVectorStoreDocumentCodec,
-        user_filter: dict[str, Any] | None = {},
-    ) -> None:
-        self.codec = codec
-        self.encoded_user_filter = (
-            codec.encode_filter(user_filter) if user_filter else {}
-        )
-
-    def encode_filter(self, filter: dict[str, Any]) -> dict[str, Any]:
-        encoded_filter = self.codec.encode_filter(filter)
-        return self._with_user_filter(encoded_filter)
+def _extract_queries(edges: set[Edge]) -> tuple[dict[str, Iterable[Any]], set[str]]:
```

Comment:
This collects the options for the queries, and returns:

```
metadata : dict[str, Iterable[any]]
  Any item with a metadata key containing a value from the iterable matches an edge.
ids : set[str]
  Any item with an `_id` matching a value in this set matches an edge.
```

Reply:
Useful to capture this in a docstring (perhaps mentioning explicitly that "the queries are intended for OR usage", so to speak)